samples/matter/common/src/fabric_table_delegate.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "app_task.h"
+
+#include <app/server/Server.h>
+#include <app/util/attribute-storage.h>
+
+namespace chip {
+
+class AppFabricTableDelegate : public FabricTable::Delegate
+{
+public:
+    ~AppFabricTableDelegate() { Server::GetInstance().GetFabricTable().RemoveFabricDelegate(this); }
+
+    /**
+     * @brief Initialize module and add a delegation to the Fabric Table.
+     *
+     * To use the OnFabricRemoved method defined within this class and allow to react on the last fabric removal
+     * this method should be called in the application code.
+     */
+    static void Init()
+    {
+#ifndef CONFIG_CHIP_LAST_FABRIC_REMOVED_NONE
+        static AppFabricTableDelegate sAppFabricDelegate;
+        Server::GetInstance().GetFabricTable().AddFabricDelegate(&sAppFabricDelegate);
+#endif // CONFIG_CHIP_LAST_FABRIC_REMOVED_NONE
+    }
+
+private:
+    void OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex)
+    {
+#ifndef CONFIG_CHIP_LAST_FABRIC_REMOVED_NONE
+        if (Server::GetInstance().GetFabricTable().FabricCount() == 0)
+        {
+#ifdef CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT
+            Server::GetInstance().ScheduleFactoryReset();
+#elif defined(CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_ONLY) || defined(CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_PAIRING_START)
+            DeviceLayer::PlatformMgr().ScheduleWork([](intptr_t) {
+                /* Erase Matter data */
+                DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().DoFactoryReset();
+                /* Erase Network credentials and disconnect */
+                DeviceLayer::ConnectivityMgr().ErasePersistentInfo();
+                DeviceLayer::ConnectivityMgr().ClearWiFiStationProvision();
```

Comment:
Are we disconnecting from Thread in this case? If so, and we want to mirror the Thread behavior to WiFi backend, we need to add `WiFiManager::Instance().Disconnect()` call to the `_ClearWiFiStationProvision()` in NrfWiFiDriver. But I'd first test if this change does not affect the WiFi operation in other places. Also, it seems to me that we do not erase the WiFi credentials in this case (ErasePersistentInfo() is implemented only for Thread, right?), so this is something we should take care of as well (we have these two keys https://github.com/nrfconnect/sdk-connectedhomeip/blob/master/src/platform/nrfconnect/wifi/NrfWiFiDriver.h#L51 stored in the KeyValueStoreMgr).

Reply:
I've added Disconnect there, and tested with nrf7002 DK, all settings are cleared so there are no mentioned keys anymore. We can perform the next commissioning of the WiFi device after the last fabric removal.